static/sass/_patterns_carousel.scss

```diff
@@ -0,0 +1,95 @@
+@mixin snapcraft-p-carousel {
```

Comment:
This looks a lot like `_snapcraft_details_screenshots.scss`. Once this lands it would be good to redo the screenshots using the carousel.

Reply:
that's what I was thinking - generalise the code and styles as it seems carousels are the future of snapcraft....